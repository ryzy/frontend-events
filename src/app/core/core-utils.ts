/**
 * Prevent re-import of core modules
 *
 * @url https://angular.io/guide/styleguide#style-04-12
 *
 * Usage:
 * ```
 * export class MyModule {
 *   constructor(@Optional() @SkipSelf() parentModule: MyModule) {
 *     throwIfAlreadyLoaded(parentModule, 'MyModule');
 *   }
 * }
 * ```
 */
export function throwIfAlreadyLoaded(parentModule: any, moduleName: string): void {
  if (parentModule) {
    throw new Error(`${moduleName} has already been loaded. Import Core modules in the AppModule only.`);
  }
}

/**
 * Returns true if code is executed in test (i.e. Karma) context.
 */
export function isUnitTestContext(): boolean {
  return !!(window as any).__karma__;
}

// tslint:disable:no-bitwise
/**
 * Generate UUID string
 */
export function uuid(length?: number): string {
  const str = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c: string) => {
    const r: number = (Math.random() * 16) | 0;
    const v: number = c === 'x' ? r : (r & 0x3) | 0x8;
    return v.toString(16);
  });

  return length ? str.substr(0, length) : str;
}
// tslint:enable:no-bitwise

/**
 * Get random number from given range
 */
export function randomRange(min: number = 0, max: number = 999): number {
  return Math.floor(Math.random() * (max - min)) + min;
}
