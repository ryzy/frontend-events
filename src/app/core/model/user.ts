import { StitchUser } from 'mongodb-stitch-browser-sdk';

/**
 * User representation in the app
 */
export class User {
  public id: string = '';
  public displayName: string = '';
  public email: string = '';
  public photoUrl: string = '';

  /**
   * Create a new instance of app User from StitchUser
   */
  public static fromStitch(user: Partial<StitchUser>): User {
    return new User({
      id: user.id || 'unknown-id',
      displayName: (user.profile && (user.profile.name || user.profile.email)) || 'User Name',
      email: (user.profile && user.profile.email) || 'unknown@email.com',
      photoUrl: (user.profile && user.profile.pictureUrl) || '/assets/default-avatar.png',
    });
  }

  public constructor(userData: Partial<User>) {
    Object.assign(this, userData);
  }
}
