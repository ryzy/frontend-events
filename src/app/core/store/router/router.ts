import { Params } from '@angular/router';
import { createFeatureSelector, createSelector, MemoizedSelector } from '@ngrx/store';
import { RouterReducerState } from '@ngrx/router-store';

/**
 * Possible URL params used in the app
 */
export interface AppRouterParams extends Params {
  eventId?: string;

  apiKey?: string; // on /user pages

  // events filters
  where?: string;
  tags?: string;
  ws?: string; // workshops
  fws?: string; // free workshops

  // sort info
  s?: string; // sort by info
  sd?: 'desc'; // sort direction. only `desc` since `asc` is by default, so it shouldn't appear in url
}

/**
 * App custom router state
 */
export interface AppRouterState {
  /**
   * Clean URL, w/o any params, query params or fragment
   */
  url: string;

  /**
   * Full URL, with whatever is present there (custom params, query params, fragment etc)
   */
  urlFull: string;

  /**
   * Any `data` added to Route config
   */
  data: { [name: string]: any };

  /**
   * Current #fragment (or empty string)
   */
  fragment: string;

  /**
   * URL :params
   */
  params: AppRouterParams;

  /**
   * URL ?query=params
   */
  queryParams: Params;
}

export const defaultAppRouterState: AppRouterState = {
  url: '/',
  urlFull: '/',
  data: {},
  fragment: '',
  params: {},
  queryParams: {},
};

export const getAppRouterState: MemoizedSelector<object, AppRouterState> = createSelector(
  createFeatureSelector<RouterReducerState<AppRouterState>>('router'),
  (state: RouterReducerState<AppRouterState>) => (state && state.state) || defaultAppRouterState,
);
