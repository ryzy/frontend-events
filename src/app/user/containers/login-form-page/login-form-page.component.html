<h2>Please log in</h2>

<form
  [formGroup]="loginForm"
  (ngSubmit)="login()"
  novalidate
  cdkTrapFocus
  class="grid-x align-center"
>
  <fieldset class="cell small-10 medium-8 large-6">
    <legend>Log in to your account</legend>
    <div class="grid-x grid-margin-x">
      <mat-form-field class="cell grow">
        <mat-label>E-mail:</mat-label>
        <input matInput name="username" formControlName="username" placeholder="email@domain.com">
      </mat-form-field>
      <mat-form-field class="cell grow">
        <mat-label>Password:</mat-label>
        <input matInput name="password" formControlName="password" type="password" placeholder="Type password here...">
      </mat-form-field>
    </div>
    <div class="cell grow">
      <button mat-raised-button color="accent" type="submit" [disabled]="!loginForm.valid">Log In</button>

      <code *ngIf="loginForm.errors && loginForm.errors['loginError']">{{loginForm.errors['loginError']}}</code>
    </div>

    <hr />

    <h6>
      Note that at this moment there is <strong>no sign-up functionality</strong> (yet).
      Use one of external providers listed on <a routerLink="/user">user page</a>.
      Get in touch if you still want to use username/password login.
    </h6>
  </fieldset>
</form>
