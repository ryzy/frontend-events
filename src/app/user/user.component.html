<h2>Your account</h2>
