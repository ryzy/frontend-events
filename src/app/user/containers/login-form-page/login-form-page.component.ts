import { Component, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { Router } from '@angular/router';
import { StitchServiceError } from 'mongodb-stitch-browser-sdk';

import { AuthService } from '../../../core/services/auth.service';

@Component({
  selector: 'app-login-form-page',
  templateUrl: './login-form-page.component.html',
  styleUrls: ['./login-form-page.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class LoginFormPageComponent {
  public loginForm: FormGroup;

  public constructor(private authService: AuthService, private router: Router, private cdRef: ChangeDetectorRef) {
    this.loginForm = new FormGroup({
      username: new FormControl('', [Validators.minLength(5)]),
      password: new FormControl('', [Validators.required]),
    });
  }

  public login(): void {
    const { username, password } = this.loginForm.getRawValue();
    if (!username || !password) {
      this.loginForm.setErrors({ loginError: 'please fill the form' });
      return;
    }

    this.authService.signInWithEmailAndPassword(username, password).subscribe(
      (user) => {
        // console.log('LoginFormPageComponent login success', { user, loginForm: this.loginForm });
        this.router.navigate(['user']);
      },
      (err: StitchServiceError) => {
        // console.warn('LoginFormPageComponent: LOGIN ERROR', { err, loginForm: this.loginForm });
        this.loginForm.setErrors({ loginError: err.message });
        this.cdRef.markForCheck();
      },
    );
  }
}
