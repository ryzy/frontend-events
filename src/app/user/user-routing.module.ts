import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

import { UserPageComponent } from './containers/user-page/user-page.component';
import { LoginFormPageComponent } from './containers/login-form-page/login-form-page.component';
import { LoginApiKeyComponent } from './containers/login-api-key/login-api-key.component';

const routes: Routes = [
  {
    path: '',
    component: UserPageComponent,
  },
  {
    path: 'login',
    component: LoginFormPageComponent,
  },
  {
    path: 'auth',
    component: LoginApiKeyComponent,
  },
  {
    path: 'auth/:apiKey',
    component: LoginApiKeyComponent,
  },
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
})
export class UserRoutingModule {}
