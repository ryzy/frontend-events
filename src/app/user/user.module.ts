import { A11yModule } from '@angular/cdk/a11y';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { MatInputModule } from '@angular/material/input';

import { SharedModule } from '../shared/shared.module';
import { UserRoutingModule } from './user-routing.module';
import { LoginApiKeyComponent } from './containers/login-api-key/login-api-key.component';
import { LoginFormPageComponent } from './containers/login-form-page/login-form-page.component';
import { UserPageComponent } from './containers/user-page/user-page.component';

@NgModule({
  imports: [SharedModule, UserRoutingModule, ReactiveFormsModule, A11yModule, MatInputModule],
  declarations: [UserPageComponent, LoginFormPageComponent, LoginApiKeyComponent],
})
export class UserModule {}
