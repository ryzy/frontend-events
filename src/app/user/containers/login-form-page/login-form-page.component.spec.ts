// tslint:disable:rxjs-throw-error
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Router } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { of, throwError } from 'rxjs';

import { mockStitchUser } from '../../../../testing/fixtures/user';
import { UserModule } from '../../user.module';
import { AppTestingAuthAndDbModule } from '../../../../testing/app-testing-auth-db.module';
import { LoginFormPageComponent } from './login-form-page.component';
import { AuthService } from '../../../core/services/auth.service';

describe('LoginFormPageComponent', () => {
  let component: LoginFormPageComponent;
  let fixture: ComponentFixture<LoginFormPageComponent>;
  let authService: AuthService;
  let router: Router;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, AppTestingAuthAndDbModule, UserModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    authService = TestBed.get(AuthService);
    router = TestBed.get(Router);
    fixture = TestBed.createComponent(LoginFormPageComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  it('should login', async () => {
    fixture.detectChanges();
    const signInSpy = spyOn(authService, 'signInWithEmailAndPassword').and.returnValue(of(mockStitchUser));
    const navigateSpy = spyOn(router, 'navigate');

    component.loginForm.setValue({
      username: 'username',
      password: 'password',
    });
    await component.login();
    fixture.detectChanges();

    expect(signInSpy).toHaveBeenCalled();
    expect(navigateSpy).toHaveBeenCalled();
  });

  it('should show login error', async () => {
    fixture.detectChanges();
    const signInSpy = spyOn(authService, 'signInWithEmailAndPassword').and.returnValue(
      throwError({ message: 'some/login/error' }),
    );
    const navigateSpy = spyOn(router, 'navigate');

    // fill the form and call login callback
    expect(component.loginForm.errors).toBeFalsy();
    component.loginForm.setValue({
      username: 'username',
      password: 'password',
    });
    await component.login();
    fixture.detectChanges();

    expect(signInSpy).toHaveBeenCalled();
    expect(navigateSpy).not.toHaveBeenCalled();

    expect((fixture.nativeElement as HTMLElement).innerText).toContain('some/login/error');
  });

  it('should do nothing when user/pass not specified', async () => {
    const signInSpy = spyOn(authService, 'signInWithEmailAndPassword').and.returnValue(of(mockStitchUser));
    await component.login();
    expect(signInSpy).not.toHaveBeenCalled();
  });
});
