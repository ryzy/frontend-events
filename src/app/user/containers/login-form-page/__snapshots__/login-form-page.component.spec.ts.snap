// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormPageComponent should create 1`] = `
<app-login-form-page
  authService={[Function AuthService]}
  cdRef={[Function ViewRef_]}
  loginForm={[Function FormGroup]}
  router={[Function Router]}
>
  <h2>
    Please log in
  </h2><div
    aria-hidden="true"
    class="cdk-visually-hidden cdk-focus-trap-anchor"
    tabindex="0"
  /><form
    cdktrapfocus=""
    class="grid-x align-center ng-untouched ng-pristine ng-invalid"
    ng-reflect-enabled=""
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <fieldset
      class="cell small-10 medium-8 large-6"
    >
      <legend>
        Log in to your account
      </legend>
      <div
        class="grid-x grid-margin-x"
      >
        <mat-form-field
          class="cell grow mat-form-field ng-tns-c0-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                  formcontrolname="username"
                  id="mat-input-0"
                  matinput=""
                  name="username"
                  ng-reflect-name="username"
                  ng-reflect-placeholder="email@domain.com"
                  placeholder="email@domain.com"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                  <label
                    aria-owns="mat-input-0"
                    class="mat-form-field-label ng-tns-c0-0 mat-empty mat-form-field-empty ng-star-inserted"
                    for="mat-input-0"
                    id="mat-form-field-label-1"
                    ng-reflect-disabled="true"
                    ng-reflect-ng-switch="true"
                  >
                    
                    
                    <mat-label
                      class="ng-star-inserted"
                    >
                      E-mail:
                    </mat-label>
                    
                    
                  </label>
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-underline ng-tns-c0-0 ng-star-inserted"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-form-field-hint-wrapper ng-tns-c0-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <mat-form-field
          class="cell grow mat-form-field ng-tns-c0-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                  formcontrolname="password"
                  id="mat-input-1"
                  matinput=""
                  name="password"
                  ng-reflect-name="password"
                  ng-reflect-placeholder="Type password here..."
                  ng-reflect-type="password"
                  placeholder="Type password here..."
                  type="password"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                  <label
                    aria-owns="mat-input-1"
                    class="mat-form-field-label ng-tns-c0-1 mat-empty mat-form-field-empty ng-star-inserted"
                    for="mat-input-1"
                    id="mat-form-field-label-3"
                    ng-reflect-disabled="true"
                    ng-reflect-ng-switch="true"
                  >
                    
                    
                    <mat-label
                      class="ng-star-inserted"
                    >
                      Password:
                    </mat-label>
                    
                    
                  </label>
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-underline ng-tns-c0-1 ng-star-inserted"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-form-field-hint-wrapper ng-tns-c0-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
      </div>
      <div
        class="cell grow"
      >
        <button
          class="mat-raised-button mat-button-base mat-accent _mat-animation-noopable"
          color="accent"
          disabled="true"
          mat-raised-button=""
          ng-reflect-color="accent"
          ng-reflect-disabled="true"
          type="submit"
        >
          <span
            class="mat-button-wrapper"
          >
            Log In
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="true"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        
      </div>
      <hr />
      <h6>
         Note that at this moment there is 
        <strong>
          no sign-up functionality
        </strong>
         (yet). Use one of external providers listed on 
        <a
          href="/user"
          ng-reflect-router-link="/user"
          routerlink="/user"
        >
          user page
        </a>
        . Get in touch if you still want to use username/password login. 
      </h6>
    </fieldset>
  </form><div
    aria-hidden="true"
    class="cdk-visually-hidden cdk-focus-trap-anchor"
    tabindex="0"
  />
</app-login-form-page>
`;

exports[`LoginFormPageComponent should show login error 1`] = `
<app-login-form-page
  authService={[Function AuthService]}
  cdRef={[Function ViewRef_]}
  loginForm={[Function FormGroup]}
  router={[Function Router]}
>
  <h2>
    Please log in
  </h2><div
    aria-hidden="true"
    class="cdk-visually-hidden cdk-focus-trap-anchor"
    tabindex="0"
  /><form
    cdktrapfocus=""
    class="grid-x align-center ng-untouched ng-pristine ng-invalid"
    ng-reflect-enabled=""
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <fieldset
      class="cell small-10 medium-8 large-6"
    >
      <legend>
        Log in to your account
      </legend>
      <div
        class="grid-x grid-margin-x"
      >
        <mat-form-field
          class="cell grow mat-form-field ng-tns-c4-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label ng-untouched ng-pristine ng-valid _mat-animation-noopable mat-form-field-should-float"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                  formcontrolname="username"
                  id="mat-input-4"
                  matinput=""
                  name="username"
                  ng-reflect-name="username"
                  ng-reflect-placeholder="email@domain.com"
                  placeholder="email@domain.com"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                  <label
                    aria-owns="mat-input-4"
                    class="mat-form-field-label ng-tns-c4-0 ng-star-inserted"
                    for="mat-input-4"
                    id="mat-form-field-label-9"
                    ng-reflect-disabled="true"
                    ng-reflect-ng-switch="true"
                  >
                    
                    
                    <mat-label
                      class="ng-star-inserted"
                    >
                      E-mail:
                    </mat-label>
                    
                    
                  </label>
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-underline ng-tns-c4-0 ng-star-inserted"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-form-field-hint-wrapper ng-tns-c4-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <mat-form-field
          class="cell grow mat-form-field ng-tns-c4-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label ng-untouched ng-pristine _mat-animation-noopable mat-form-field-should-float ng-valid"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                  formcontrolname="password"
                  id="mat-input-5"
                  matinput=""
                  name="password"
                  ng-reflect-name="password"
                  ng-reflect-placeholder="Type password here..."
                  ng-reflect-type="password"
                  placeholder="Type password here..."
                  type="password"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                  <label
                    aria-owns="mat-input-5"
                    class="mat-form-field-label ng-tns-c4-1 ng-star-inserted"
                    for="mat-input-5"
                    id="mat-form-field-label-11"
                    ng-reflect-disabled="true"
                    ng-reflect-ng-switch="true"
                  >
                    
                    
                    <mat-label
                      class="ng-star-inserted"
                    >
                      Password:
                    </mat-label>
                    
                    
                  </label>
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-underline ng-tns-c4-1 ng-star-inserted"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-form-field-hint-wrapper ng-tns-c4-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
      </div>
      <div
        class="cell grow"
      >
        <button
          class="mat-raised-button mat-button-base mat-accent _mat-animation-noopable"
          color="accent"
          disabled="true"
          mat-raised-button=""
          ng-reflect-color="accent"
          ng-reflect-disabled="true"
          type="submit"
        >
          <span
            class="mat-button-wrapper"
          >
            Log In
          </span>
          <div
            class="mat-button-ripple mat-ripple"
            matripple=""
            ng-reflect-centered="false"
            ng-reflect-disabled="true"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        
        <code
          class="ng-star-inserted"
        >
          some/login/error
        </code>
      </div>
      <hr />
      <h6>
         Note that at this moment there is 
        <strong>
          no sign-up functionality
        </strong>
         (yet). Use one of external providers listed on 
        <a
          href="/user"
          ng-reflect-router-link="/user"
          routerlink="/user"
        >
          user page
        </a>
        . Get in touch if you still want to use username/password login. 
      </h6>
    </fieldset>
  </form><div
    aria-hidden="true"
    class="cdk-visually-hidden cdk-focus-trap-anchor"
    tabindex="0"
  />
</app-login-form-page>
`;
